src/components/Button/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { Loading } from '@components/Loading'
+import { HTMLAttributes } from 'react'
+import styled, { css, DefaultTheme } from 'styled-components'
+
+export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  children: string
+  size: 'large' | 'medium'
+  disabled?: boolean
+  isLoading?: boolean
+  block: boolean
+}
+
+const sizeModifiers = {
+  medium: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.large};
+    font-size: ${theme.fonts.sizes.large};
+  `,
+  large: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.xlarge};
+    font-size: ${theme.fonts.sizes.xlarge};
+  `,
+}
+
+const Wrapper = styled.button<Omit<ButtonProps, 'children'>>`
+  ${({ theme, size, disabled, block }) => css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: ${block ? '100%' : 'auto'};
+    padding: 0 1rem;
+    color: ${theme.colors.white};
+    background-color: ${theme.colors.black};
+    font-weight: bold;
+    border-radius: 0.2rem;
+    cursor: pointer;
+    ${size && sizeModifiers[size](theme)};
+    ${disabled &&
+    css`
+      opacity: 0.3;
+      pointer-events: none;
+    `}
+    :hover {
+      background-color: ${theme.colors.lightBlack};
+    }
+    :focus {
+      border: 0.2rem solid ${theme.colors.grey};
+    }
+  `};
+`
+
+const Button = ({
+  children,
+  size,
+  disabled,
+  isLoading,
+  block,
```

Comment:
Faltou dar spread nas demais props como onClick por exemplo.

Reply:
Fix in this commit [2379b1](https://github.com/joaooliveiradev/conduit-frontend/commit/2379b15a26811166ff6a96e17d37e93fe77e06e6)